maven-embedder/src/main/java/org/apache/maven/cli/transfer/AbstractMavenTransferListener.java

```diff
@@ -214,18 +227,37 @@ public void transferSucceeded(TransferEvent event) {
         FileSizeFormat format = new FileSizeFormat(Locale.ENGLISH);
 
         StringBuilder message = new StringBuilder();
-        message.append(action).append(' ').append(direction).append(' ').append(resource.getRepositoryId());
-        message.append(": ");
-        message.append(resource.getRepositoryUrl()).append(resource.getResourceName());
-        message.append(" (").append(format.format(contentLength));
+        message.append(action).append(lowOn).append(' ').append(direction).append(' ');
+        message.append(lowOff).append(resource.getRepositoryId());
+        message.append(lowOn).append(": ");
+        message.append(resource.getRepositoryUrl()).append(lowOff);
+        appendResource(message, resource.getResourceName());
+        message.append(lowOn).append(" (").append(format.format(contentLength));
 
         long duration = System.currentTimeMillis() - resource.getTransferStartTime();
         if (duration > 0L) {
             double bytesPerSecond = contentLength / (duration / 1000.0);
             message.append(" at ").append(format.format((long) bytesPerSecond)).append("/s");
         }
 
-        message.append(')');
+        message.append(')').append(lowOff);
         out.println(message.toString());
     }
+
+    private void appendResource(StringBuilder message, String resource) {
+        if (!MessageUtils.isColorEnabled()) {
+            message.append(resource);
+            return;
+        }
+        int filename = resource.lastIndexOf('/');
+        int version = resource.substring(0, filename).lastIndexOf('/');
+        int artifactId = resource.substring(0, version).lastIndexOf('/');
```

Comment:
Can we fully make this assumption?

Reply:
AFAIK, it is based on repository layout: https://maven.apache.org/repositories/layout.html

then I don't see what can happen in this code that would not fully break the repository

Reply:
Then add a comment and link to the resource and that we assume only the default Maven repository layout. If someone, does custom stuff with a custom repo format, e.g., p2, then this listener will be broken.

Reply:
thinking twice: there is an edge case on `maven-metadata.xml`, that can be also at groupId or atifactId level
![MNG-7875_maven-medata](https://github.com/apache/maven/assets/237462/c9144450-d4cd-4e44-aab6-44836b3f2501)
I'll need to add a test for that, not a problem

and yes, I'll add some more comments and links

Reply:
@laeubi is that transfer message used by Tycho?

Reply:
testing on https://github.com/eclipse-tycho/tycho/tree/master/demo/itp04-rcp

```
❯ \rm -rf ~/.m2/repository/p2/
❯ mvn package
[INFO] Scanning for projects...
[INFO] Resolving target definition file:/home/herve/tmp/tycho/demo/itp04-rcp/mars/mars.target for environments=[win32/win32/x86_64, linux/gtk/x86_64, macosx/cocoa/x86_64], include source mode=honor, execution environment=StandardEEResolutionHints [executionEnvironment=OSGi profile 'JavaSE-11' { source level: 11, target level: 11}], remote p2 repository options=org.eclipse.tycho.p2.remote.RemoteAgent@77db231c...
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[INFO] Resolving dependencies of MavenProject: example.group:eclipse-repository:1.0.0-SNAPSHOT @ /home/herve/tmp/tycho/demo/itp04-rcp/eclipse-repository/pom.xml
[INFO] Resolving dependencies of MavenProject: example.group:example-bundle:0.1.0-SNAPSHOT @ /home/herve/tmp/tycho/demo/itp04-rcp/example-bundle/pom.xml
[INFO] Fetching 202206151000&countryCode=fr&timeZone=1&format=xml from https://www.eclipse.org/downloads/download.php?format=xml&file=/releases/2022-06/
[INFO] Fetching com.ibm.icu_67.1.0.v20200706-1749.jar from https://mirror.aarnet.edu.au/pub/eclipse/releases/2022-06/202206151000/plugins/ (12,94MB)
[INFO] Fetching com.ibm.icu_67.1.0.v20200706-1749.jar from https://mirror.aarnet.edu.au/pub/eclipse/releases/2022-06/202206151000/plugins/ (179,24kB of 12,94MB at 175,24kB/s)
[INFO] Fetching com.ibm.icu_67.1.0.v20200706-1749.jar from https://mirror.aarnet.edu.au/pub/eclipse/releases/2022-06/202206151000/plugins/ (1,67MB of 12,94MB at 782,9kB/s)
[INFO] Fetching com.ibm.icu_67.1.0.v20200706-1749.jar from https://mirror.aarnet.edu.au/pub/eclipse/releases/2022-06/202206151000/plugins/ (6,32MB of 12,94MB at 2,1MB/s)
...
```

transfer from p2 seems to be implemented outside Maven core, then not affected

assumption of Maven 2 repository layout seems reasonable

Reply:
Layout is delegated to resolver. since Maven3.0. Maven cannot and should not make any assumptions here, all Maven can do is to control what layout is to be used with resolver. Doing assumptions like these is plain wrong.